src/tests/banners/bannerSelection.tsx

```diff
@@ -34,70 +49,150 @@ const fetchBannerDeployTime = (region: ReaderRevenueRegion) => (): Promise<Date>
+            fiveMinutes,
+            'fetchSubscriptionsBannerDeployTime_european-union',
+        ),
+    },
 };
 
 // Has the banner been redeployed since the user last closed it?
-const redeployedSinceLastClosed = (targeting: BannerTargeting): Promise<boolean> => {
-    if (targeting.engagementBannerLastClosedAt) {
-        const closedAt = targeting.engagementBannerLastClosedAt;
-        const region = readerRevenueRegionFromCountryCode(targeting.countryCode);
-
-        const [, getCached] = caches[region];
+export const redeployedSinceLastClosed = (
+    targeting: BannerTargeting,
+    bannerType: BannerType,
+): Promise<boolean> => {
+    const { subscriptionsBannerLastClosedAt, engagementBannerLastClosedAt } = targeting;
 
-        return getCached().then(deployDate => deployDate > new Date(closedAt));
-    } else {
+    if (
+        (bannerType === 'subscriptions' &&
+            typeof subscriptionsBannerLastClosedAt === 'undefined') ||
+        (bannerType === 'contributions' && typeof engagementBannerLastClosedAt === 'undefined')
+    ) {
         return Promise.resolve(true);
     }
+
+    const region = readerRevenueRegionFromCountryCode(targeting.countryCode);
+
+    if (bannerType === 'subscriptions') {
+        const [, getCached] = caches.subscriptions[region];
+        return getCached().then(deployDate => {
+            return (
+                !subscriptionsBannerLastClosedAt ||
+                deployDate > new Date(subscriptionsBannerLastClosedAt)
+            );
+        });
+    } else if (bannerType === 'contributions') {
+        const [, getCached] = caches.contributions[region];
+        return getCached().then(deployDate => {
+            return (
+                !engagementBannerLastClosedAt || deployDate > new Date(engagementBannerLastClosedAt)
+            );
+        });
+    }
+
+    return Promise.resolve(true);
 };
 
 // TODO - implement test selection properly
-export const selectBannerTest = (
+export const selectBannerTest = async (
     targeting: BannerTargeting,
     pageTracking: BannerPageTracking,
     baseUrl: string,
 ): Promise<BannerTestSelection | null> => {
-    return redeployedSinceLastClosed(targeting).then(hasRedeployed => {
-        if (hasRedeployed) {
-            const tests = [AusMomentContributionsBanner];
-            const testToRun = tests.find(
-                test =>
-                    targeting.alreadyVisitedCount >= test.minPageViews &&
-                    !targeting.shouldHideReaderRevenue &&
-                    !targeting.isPaidContent &&
-                    test.canRun(targeting, pageTracking),
-            );
+    const tests: BannerTest[] = [
+        AusMomentContributionsBanner,
+        DigitalSubscriptionBanner,
+        WeeklyBanner,
+    ];
+
+    for (const test of tests) {
+        if (
+            !targeting.shouldHideReaderRevenue &&
+            !targeting.isPaidContent &&
+            targeting.showSupportMessaging &&
```

Comment:
For contributions there are cases where we _do_ want to show something to supporters, and this check against `showSupportMessaging` will prevent that from happening.
E.g. the current Aus moment banner is seen by all.

My preferred solution here would be to add a field to the `BannerTest` type, e.g.:
`type BannerAudience = 'NonSupporters' | 'Supporters' | 'All';`

Then something like:
```
const audienceMatches = (showSupportMessaging: boolean, testAudience: BannerAudience): boolean => {
    switch (testAudience) {
        case 'NonSupporters':
            return showSupportMessaging;
        case 'Supporters':
            return !showSupportMessaging;
        default:
            return true;
    }
};
```

Reply:
Great point! Thanks Tom 👍